storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -1746,85 +1851,19 @@ copy_back()
 
 	it = datadir_iter_new(".", false);
```

Comment:
It is not true for backup stage. When backing up, we copy redo logs in separate thread. We also build fli_space dictionary and ibdata1 is just one of the tablespaces, it is copied in parallel with file-per-table tablespaces. Same is true for undo tablesaces. Your implementation is fine for copy-back. I just wanted to explain how backup works.